integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -492,6 +537,38 @@ fn test_resharding_v3() {
     test_resharding_v3_base(TestReshardingParametersBuilder::default().build());
 }
 
+// Takes a sequence of shard ids to track in consecutive epochs,
+// repeats the last element `TESTLOOP_NUM_EPOCHS_TO_WAIT` times,
+// and maps each element: |id| -> vec![ShardId(id)], to the format required by `TrackedShardSchedule`.
+fn shard_sequence_to_schedule(mut shard_sequence: Vec<u64>) -> Vec<Vec<ShardId>> {
+    shard_sequence.extend(
+        std::iter::repeat(*shard_sequence.last().unwrap())
+            .take(TESTLOOP_NUM_EPOCHS_TO_WAIT as usize),
+    );
+    shard_sequence.iter().map(|shard_id| vec![ShardId::new(*shard_id)]).collect()
+}
+
+#[test]
+// TODO(resharding): fix nearcore and un-ignore this test
+#[ignore]
+fn test_resharding_v3_state_cleanup() {
+    // Track parent shard (6) before resharding, child shard (7) after resharding, and then an unrelated shard (5) forever.
+    // Eventually, the State column should only contain entries belonging to shard 5.
+    let tracked_shard_sequence = vec![6, 6, 7, 5];
```

Comment:
I'm wondering if we should try to assert that resharding indeed happens at the wanted epoch (the third?). Maybe not worth the effort.

Reply:
I plan more tests with different tracked shard schedule, not necessarily tracking a child shard in the epoch after resharding.
But will add a check for each epoch that it has the corresponding shard in its layout.

Reply:
fixed in 66f279e879863ae6b1144cdd6ba70c595c62cf6c